cl/corpus_importer/management/commands/import_tn.py

```diff
@@ -0,0 +1,188 @@
+import itertools
+import json
+import logging
+from glob import glob
+
+from dateutil import parser
+from django.utils.encoding import force_bytes
+
+from cl.lib.command_utils import VerboseCommand
+from cl.lib.crypto import sha1
+from cl.people_db.models import Person
+from cl.scrapers.management.commands.cl_scrape_opinions import (
+    make_objects,
+    save_everything,
+)
+from cl.scrapers.tasks import extract_doc_content
+from cl.search.models import Court, Docket, Opinion
+
+
+def make_item(case):
+    """Make an import item for our save everything function
+
+    :param case: Case data received from the court.
+    :return: Case information.
+    :type: dict
+    """
+
+    judges = case["judge"]
+    first_name = judges.split(" ")[0].title()
+
+    lead_author = Person.objects.get(
+        positions__court_id=case["court"],
+        is_alias_of=None,
+        name_first=first_name,
+    )
+    # Four Panelists exist, one retired end of 2019 and the other joined.
+    panelists = case["judge"]
+    if case["court"] == "tennworkcompapp":
+        exclude = (
+            "Marshall"
+            if parser.parse(case["pub_date"]).date().year == 2020
+            else "Pele"
+        )
+        panelists_query = Person.objects.filter(
+            positions__court_id=case["court"]
+        ).exclude(name_first=exclude)
+        panelists = ", ".join([x.name_full for x in panelists_query])
+
+    return {
+        "source": Docket.DIRECT_INPUT,
+        "cluster_source": "D",
+        "case_names": case["title"],
+        "case_dates": parser.parse(case["pub_date"]),
+        "precedential_statuses": "Published",
+        "docket_numbers": case["docket"],
+        "judges": panelists,
+        "author_id": lead_author.id,
+        "author": lead_author,
+        "date_filed_is_approximate": False,
+        "blocked_statuses": False,
+        "neutral_citations": case["neutral_citation"],
+        "download_urls": case["pdf_url"],
+    }
+
+
+def add_neutral_citations(filepath):
+    """Add neutral citations to our dataset
+
+    :param tn_corpus: The case data.
+    :type: dict
+    :return: Our case data with neutral citations included.
+    :type: dict
+    """
+    tn_corpus = json.loads(open(filepath, "r").read())
+    results = []
+    for court in ["tennworkcompapp", "tennworkcompcl"]:
+        reporter = "TN WC App." if court == "tennworkcompapp" else "TN WC"
+        tn = [
+            x
+            for x in sorted(
+                tn_corpus, key=lambda x: (x["pub_date"], x["label"])
+            )
+            if x["court"] == court
+        ]
+        for key, group in itertools.groupby(tn, lambda x: x["pub_date"][:4]):
+            count = 1
+            for case in list(group):
+                case["neutral_citation"] = "%s %s %s" % (key, reporter, count)
+                results.append(case)
+                count += 1
+    return results
```

Comment:
This function does...a lot that isn't adding neutral citations. Seems like you should have one function that opens the data and returns it, and a second that loops over that data and adds neutral citations. 

I also find the logic here very strange:

 - for each of the two courts
 - loop through the entire data set to filter it
 - add the filtered data to a var if it matches the court name we're working on
 - sort and group the data and do neutral citations

Why not just loop through the data once to organize it (in one function called from `import_tn_corpus`, and then have this function just add neutral citations? Should be much clearer.

Reply:
just yanked the whole function.  No need to dynamically generate what I can just add to our data set.